packages/commonwealth/client/scripts/views/pages/CommunityManagement/Topics/CreateTopicsSection/CreateTopicSection.tsx

```diff
@@ -91,45 +86,41 @@ export const CreateTopicSection = () => {
         'char',
       )}
     ${disallowedCharMatches.join(', ')} are not permitted`;
-      setErrorMsg(err);
+      setNameErrorMsg(err);
       return ['failure', err];
     }
 
-    if (errorMsg) {
-      setErrorMsg(null);
-    }
+    setNameErrorMsg(null);
 
     return ['success', 'Valid topic name'];
   };
 
+  // TODO: implement logic around creating a new thread with new topic
   return (
     <div className="CreateTopicSection">
-      <div className="form">
+      <CWForm
+        validationSchema={topicCreationValidationSchema}
+        onSubmit={handleCreateTopic}
+      >
         <div className="form-inputs">
-          <div>
-            <CWTextInput
-              label="Name (required)"
-              placeholder="Enter a topic name"
-              value={name}
-              onInput={(e) => {
-                setName(e.target.value);
-              }}
-              inputValidationFn={handleInputValidation}
-              autoFocus
-            />
-            {errorMsg && (
-              <CWValidationText
-                className="validation-text"
-                message={errorMsg}
-                status="failure"
-              />
-            )}
-          </div>
-
           <CWTextInput
+            hookToForm
+            label="Name (required)"
+            placeholder="Enter a topic name"
+            name="topicName"
+            value={name}
+            onInput={(e) => {
+              setName(e.target.value);
+              handleInputValidation(e.target.value);
```

Comment:
I still think we can remove the `handleInputValidation` validation. It looks like the function does these validations
- duplicate topic name checks (which we should keep)
- check for unallowed chars -- which we can remove and use the zod validation for topicName instead, ex:
``` 
topicName: z
    .string({ invalid_type_error: VALIDATION_MESSAGES.NO_INPUT }).refine((value) => {
// check validation here with js logic
// return bool to indicate success/failure for validation, and it will print appropriate error msg or you can add custom msg
})
```

Reply:
Disallowed characters now handled with Zod!